tests/dbshell/tests.py

```diff
@@ -15,3 +16,25 @@ def test_command_missing(self):
         with self.assertRaisesMessage(CommandError, msg):
             with mock.patch('subprocess.run', side_effect=FileNotFoundError):
                 call_command('dbshell')
+
+
+class BaseDatabaseClientTestCase(SimpleTestCase):
```

Comment:
IMHO the test adds no value, the runshell function is 3 lines long and now explicitly set env to None when needed.

Reply:
> The test adds value as it ensures that this doesn't regress again in the future.

Unit-testing absolutist would refactor that function into something like this:

```python
    def runshell(self, parameters):
        args, env = self.settings_to_cmd_args_env(self.connection.settings_dict, parameters)
        env = self._merge_env(env, os.environ)
        subprocess.run(args, env=env, check=True)
        
    def _merge_env(env, os_env):
         return {**os.environ, **env} if env else None
```

and just test _merge_env, but does it make much sense?

> In addition, it helps database backends implemented in external packages match the expected behaviour.
 
I don't see how that test could help with new backends, the test is local to BaseDatabaseClient and tests one line
```python
env = {**os.environ, **env} if env else None
```